### PR TITLE
vdaf: Update API to match syntax

### PR DESCRIPTION
**Pull request by cjpatton**
See https://cjpatton.github.io/vdaf/draft-patton-cfrg-vdaf.html.

### CODE REVIEW

File: src/vdaf.rs [hunk 1]
```diff
@@ -256,88 +282,43 @@ pub struct VerifierMessage<F> {
...
-                "verify_start(): called on incorrect state: {:?}",
-                state
-            )));
-        }
-    };
+) -> Result<(VerifyState<V>, VerifyMessage<V::Field>), VdafError> {
+    let mut deriver = KeyDeriver::from_key(&verify_param.query_rand_init);
+    deriver.update(&[255]);
+    deriver.update(nonce);
+    let query_rand_seed = deriver.finish();
```
Comment (tgeoghegan) [inline src/vdaf.rs:310]: I'm checking this against the `vdaf_start` definition for Prio3:
```
  k_query_rand = get_key(k_query_init, nonce)
  (...)
  query_rand = expand(k_query_rand, QUERY_RAND_LEN)
```
So, I see how `k_query_init` is `verify_param.query_rand_init`, and `nonce` is `nonce`, but where does the `&[255]` fed into the deriver come from?

File: src/vdaf.rs [hunk 2]
```diff
@@ -430,18 +400,33 @@ where
...
+    Ok(state.input_share)
+}
+
+fn check_num_shares(func_name: &str, num_shares: u8) -> Result<(), VdafError> {
+    if num_shares == 0 {
+        return Err(VdafError::Uncategorized(format!(
+            "{}(): at least one share is required; got {}",
+            func_name, num_shares
+        )));
+    } else if num_shares > 254 {
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:423]: The high byte (255) is reserved for deriving the query randomness seed from the nonce.

File: src/vdaf.rs [hunk 3]
```diff
@@ -121,17 +121,12 @@ impl HelperShare {
 /// * `suite` is the cipher suite used for key derivation.
 /// * `input` is the input to be secret shared.
 /// * `num_shares` is the number of input shares (i.e., aggregators) to generate.
-pub fn dist_input<V: Value>(
+pub fn vdaf_input<V: Value>(
     suite: Suite,
     input: &V,
     num_shares: u8,
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:127]: Done.

File: src/vdaf.rs [hunk 4]
```diff
@@ -240,10 +235,41 @@ pub fn dist_input<V: Value>(
     Ok(out)
 }
 
+/// The verification parameter used by each aggregator to evaluate the VDAF.
+#[derive(Clone, Debug)]
+pub struct VerifyParam<V: Value> {
+    /// Input parameter, needed to reconstruct a [`Value`] from a vector of field elements.
+    pub input_param: V::Param,
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:242]: Good call, done.

File: src/vdaf.rs [hunk 5]
```diff
@@ -240,10 +235,41 @@ pub fn dist_input<V: Value>(
...
+    /// The identity of the aggregator.
+    pub aggregator_id: u8,
+}
+
+#[cfg(test)]
+fn vdaf_setup<V: Value>(
+    suite: Suite,
+    input_param: &V::Param,
+    num_shares: u8,
+) -> Result<Vec<VerifyParam<V>>, VdafError> {
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:256]: Done.

File: src/vdaf.rs [hunk 6]
```diff
@@ -256,88 +282,43 @@ pub struct VerifierMessage<F> {
...
 /// that will be broadcast to the other aggregators.
-// TODO(cjpatton) Check for ciphersuite mismatch between `state` and `msg`.
-pub fn verify_start<V: Value>(
-    state: AggregatorState<V>,
+//
+// TODO(cjpatton) Check for ciphersuite mismatch between `verify_param` and `msg`.
+pub fn vdaf_start<V: Value>(
+    verify_param: &VerifyParam<V>,
+    nonce: &[u8],
     msg: InputShareMessage<V::Field>,
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:305]: Done.

File: src/vdaf.rs (hunk 1 above)
Comment (cjpatton) [reply src/vdaf.rs:310]: Good catch, updated the spec.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**